airflow/models.py

```diff
@@ -1183,13 +1183,19 @@ def next_retry_datetime(self):
         """
         delay = self.task.retry_delay
         if self.task.retry_exponential_backoff:
+            min_backoff = int(delay.total_seconds() * (2 ** (self.try_number - 2)))
+            # deterministic per task instance
+            hash = int(hashlib.sha1("{}#{}#{}".format(self.dag_id, self.task_id, self.execution_date)
```

Comment:
Not strictly necessary, but I think separators like this are often used when hashing concatted stirngs that are different to disambiguate cases where task_id + dag_id = task_id2 + dag_id2. Not that it makes a huge difference here.